responses/__init__.py

```diff
@@ -469,7 +471,9 @@ def _url_matches(self, url: "_URLPatternType", other: str) -> bool:
             if _has_unicode(url):
                 url = _clean_unicode(url)
 
-            return _get_url_and_path(url) == _get_url_and_path(other)
+            return _get_url_and_path(url) == _get_url_and_path(
+                _ensure_url_default_path(other)  # type: ignore[arg-type]
```

Comment:
To explain this change:

Requests, in "prepare_url" does things that end to "enforce" a trailing "/". Which isn't enforced by Niquests.

So, we have to enforce consistency across both library. here, only Niquests is affected. Not Requests.